source/common/upstream/outlier_detection_impl.cc

```diff
@@ -330,24 +408,28 @@ void DetectorImpl::onConsecutiveErrorWorker(HostSharedPtr host, EjectionType typ
 
   // We also reset the appropriate counter here to allow the monitor to detect a bout of consecutive
   // error responses even if the monitor is not charged with an interleaved non-error code.
+  updateDetectedEjectionStats(type);
+  ejectHost(host, type);
+
+  // reset counters
   switch (type) {
   case EjectionType::Consecutive5xx:
     stats_.ejections_consecutive_5xx_.inc(); // Deprecated
-    stats_.ejections_detected_consecutive_5xx_.inc();
-    ejectHost(host, EjectionType::Consecutive5xx);
     host_monitors_[host]->resetConsecutive5xx();
     break;
   case EjectionType::ConsecutiveGatewayFailure:
-    stats_.ejections_detected_consecutive_gateway_failure_.inc();
-    ejectHost(host, EjectionType::ConsecutiveGatewayFailure);
     host_monitors_[host]->resetConsecutiveGatewayFailure();
     break;
-  case EjectionType::SuccessRate:
+  case EjectionType::ConsecutiveLocalOriginFailure:
+    host_monitors_[host]->resetConsecutiveLocalOriginFailure();
+    break;
+  case EjectionType::SuccessRateExternalOrigin:
+  case EjectionType::SuccessRateLocalOrigin:
     NOT_REACHED_GCOVR_EXCL_LINE;
   }
 }
 
-Utility::EjectionPair Utility::successRateEjectionThreshold(
+std::tuple<double, double> DetectorImpl::successRateEjectionThreshold(
```

Comment:
Any particular reason why you are using `std::tuple` rather than `std::pair` for this two element tuple? In general I think both approaches suffer a bit from being opaque about what the pair of numbers is, maybe a typedef or simple struct is appropriate? What do you think?

Reply:
Converted to use simple struct.